### PR TITLE
staging-next 2024-09-21

### PR DESCRIPTION
**Pull request by vcunat**
EDIT: now contains changes up to 09-24.

##### Helpful links

- workflow docs: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#staging
- important jobs:
  https://hydra.nixos.org/job/nixpkgs/staging-next/unstable#tabs-constituents
  https://hydra.nixos.org/job/nixos/staging-next-small/tested
- jobsets:
  https://hydra.nixos.org/jobset/nixpkgs/staging-next
  https://hydra.nixos.org/jobset/nixos/staging-next-small
- nix-review reports: https://malob.github.io/nix-review-tools-reports/
- previous staging-next: https://github.com/NixOS/nixpkgs/pull/338840
##### Mass breakages
_nothing notable so far_

### CODE REVIEW

**Comment by fabianhjr** [comment]
https://github.com/NixOS/nixpkgs/pull/343437

**Comment by GaetanLepage** [comment]
> previous staging-next:

I think that the description mentions the wrong previous staging-next. The true previous one is https://github.com/NixOS/nixpkgs/pull/338840.

**Comment by vcunat** [comment]
Correct, thanks.  I forgot to edit that after copy&paste.  Fixed now.

**Comment by fabianhjr** [comment]
Will wait on hydra but seems like nodejs might not pass test phase again after #343683 

```
       > Failed tests:
       > out/Release/node /build/node-v20.17.0/test/parallel/test-tls-client-renegotiation-13.js
       > out/Release/node /build/node-v20.17.0/test/parallel/test-tls-client-verify.js
       > out/Release/node /build/node-v20.17.0/test/parallel/test-tls-multiple-cas-as-string.js
       > out/Release/node /build/node-v20.17.0/test/parallel/test-tls-multi-key.js
       > out/Release/node /build/node-v20.17.0/test/parallel/test-tls-peer-certificate-encoding.js
       > out/Release/node /build/node-v20.17.0/test/parallel/test-tls-sni-server-client.js
       > make: *** [Makefile:555: test-ci-js] Error 1
```

**Comment by NickCao** [comment]
`protobuf_28` breaks cross compilation,  can be fixed by fetching https://github.com/protocolbuffers/protobuf/pull/17888

**Comment by vcunat** [comment]
- [ ] `python3Packages.mlflow` didn't survive update of the default protobuf
  maintainer: @tbenst
  https://hydra.nixos.org/build/274530357/nixlog/1/tail

**Comment by GaetanLepage** [comment]
> `python3Packages.mlflow` didn't survive update of the default protobuf
> maintainer: @tbenst
> https://hydra.nixos.org/build/274530357/nixlog/1/tail

Handled in #347104

**Comment by andrevmatos** [comment]
I think `amdvlk` is broken in staging-next, and may require #344568

File: pkgs/development/python-modules/certbot/default.nix [hunk 1]
```diff
@@ -33,7 +34,16 @@ buildPythonPackage rec {
     hash = "sha256-Qee7lUjgliG5fmUWWPm3MzpGJHUF/DXZ08UA6kkWjjk=";
   };
 
-  sourceRoot = "${src.name}/${pname}";
+  patches = [
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/certbot/default.nix:37]: Next time just drop the pytest.ini instead. The settings in there just make any new warning an error which is for the developers, not for us.

**Pull request merged**